app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -36,6 +54,7 @@ class MetadataForm {
     this.fields = fields
     this.values[fields.metadataKey.id] = formData[fields.metadataKey.name]
     this.values[fields.metadataValue.id] = formData[fields.metadataValue.name]
+    this.existingMetadata = formData['existing-metadata']
```

Comment:
I think I'd recommend passing this in as a separate (optional) parameter to the constructor. As existing metadata is not part of the actual form data this would keep the process clear

Something like
```js
class MetadataForm {
  constructor(formData = {}, existingMetadata = {}) {
    // ...
    this.existingMetadata = existingMetdata
  }
}
```

Reply:
Done.